nilearn/tests/test_numpy_conversions.py

```diff
@@ -230,3 +231,15 @@ def test_as_ndarray():
     ## Unhandled cases
     assert_raises(ValueError, as_ndarray, "test string")
     assert_raises(ValueError, as_ndarray, [], order="invalid")
+
+
+def test_csv_to_array():
+    # Create a phony CSV file
+    with tempfile.NamedTemporaryFile(suffix='.csv', delete=False,
+                                     mode='wt') as csv_file:
+        csv_file.write('1.,2.,3.,4.,5.\n')
```

Comment:
> Yes. For some reason, it doesn't work if I don't close the file, I don't know why :-/

Flushing the file ? csv_file.flush() seems to do it for me and you don't need the delete=False.